### DISCUSSION TITLE
Any reason why goto takes invalidateAll but not invalidate?

### DISCUSSION THREAD

**Discussion by gterras** [sveltejs/kit#10655] [Q&A] [answered]
It would be nice to be able to invalidate a custom identifier on goto(). Is there a reason for this? Or any other way to do so? Thanks!

**Comment by teemingc** [comment]
A solution is currently being discussed here https://github.com/sveltejs/kit/issues/6294#issuecomment-1685052381

**Reply by gterras** [reply]
Thanks for your answer, may I ask if it would be worth it to isolate the `goto takes invalidate` issue from `Opt out of load invalidation` that seems quite a bigger issue? 

My use case feels quite common, in a simple CRUD with the following pages : 
- /list 
- /list/[id]

Say you have a `delete` button on the `[id]` page that calls a `DELETE` action at the same level, if you `invalidate` then `goto`  the `[id]` load function will get a not found (obviously). You can solve this by redirecting to the parent page on error in the load function but this doesn't feel right (one extra -failed- request per deletion). I know I could redirect to some dummy invalidation page or add a query parameter that would trigger an invalidation in the `list` page but it feels complicated.  

`goto(invalidateAll)` won't really work here if you have unrelated parametrized load functions in previous segments of the URL (ie if you're using `depends` anywhere).

**Reply by teemingc** [reply, marked as answer]
> Thanks for your answer, may I ask if it would be worth it to isolate the `goto takes invalidate` issue from `Opt out of load invalidation` that seems quite a bigger issue? 

Yes, you definitely could.

**Reply by gterras** [reply]
Thanks for your help I've opened an issue here https://github.com/sveltejs/kit/issues/10659
